core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -197,9 +197,22 @@ static StreamMessage<HttpData> of(File file) {
      * Therefore, the returned {@link StreamMessage} will emit {@link HttpData}s chunked to
      * size less than or equal to {@value PathStreamMessage#DEFAULT_FILE_BUFFER_SIZE}.
      */
-    static StreamMessage<HttpData> of(Path path) {
+    static ByteStreamMessage of(Path path) {
         requireNonNull(path, "path");
-        return of(path, DEFAULT_FILE_BUFFER_SIZE);
+        return of(path, null);
+    }
+
+    /**
+     * Creates a new {@link StreamMessage} that streams the specified {@link Path}.
+     * The specified {@code bufferSize} is used to create a buffer used to read data from the {@link Path}.
+     * Therefore, the returned {@link StreamMessage} will emit {@link HttpData}s chunked to
+     * size less than or equal to {@code bufferSize}.
+     *
+     * @param path the path of the file
+     * @param executor the {@link ExecutorService} which performs blocking IO read
+     */
+    static ByteStreamMessage of(Path path, @Nullable ExecutorService executor) {
```

Comment:
How about adding `StreamMessage.builder(path)`? e.g.
```
ByteStreamMessage stream =
  StreamMessage
    .builder(path)
    .bufferSize(...)
    .alloc(...)
    .executor(...)
    .range(1048576, 768) // skip 1MiB and read 768 bytes
    .build()
```

Reply:
It is hard to apply `range()`, bufferSize()` with the builder pattern to the `ByteStreamMessage` created from `HttpResponse.split().body()`.
Do you think we need to support both builder and member functions?

Reply:
Commented on https://github.com/line/armeria/pull/4058#discussion_r895475244